runtime/runtime_test.go

```diff
@@ -7236,3 +7237,496 @@ func TestRuntimeComputationMetring(t *testing.T) {
 		})
 	}
 }
+
+func TestRuntimeResourcePrecondition(t *testing.T) {
```

Comment:
Could these be just interpreter tests, or do they also rely on behaviour in the runtime package?
If not, please move them to the interpreter tests, which reduces the amount of boilerplate.

Could you please also test two additional scenarios:

- Test with non-resource kinded arguments (e.g. by adding an additional non-resource-kinded parameter, so we have coverage for the `continue` branch in the for-loop which temporarily untracks resources in the condition function wrapper, above)
- Test with multiple interfaces, so that multiple wrappings with temporary untracking and restoring of resource-kinded arguments is tested

Reply:
I think https://github.com/onflow/cadence/pull/1625/files#diff-74ffa81b6938856f695ba112a721503dfc5ff2e852bc65d79d15ee4868f4bb97R7598 should test multiple wrappings of functions, but let me know if you had something else in mind.

Reply:
Ah,  `TestInterpreterResourceDoubleWrappedCondition` looks good 👍